src/v/cluster/partition_balancer_state.h

```diff
@@ -47,6 +47,8 @@ class partition_balancer_state {
       const std::vector<model::broker_shard>& prev,
       const std::vector<model::broker_shard>& next);
 
+    ss::future<> apply_snapshot(const controller_snapshot&);
```

Comment:
I am wondering why do we need to keep the balancer state in the snapshot now ? Previously the state wasn't persistent

Reply:
The balancer needs a list of ntps with violated rack constraints that is maintained by applying topic table commands. When applying a snapshot, we need to rebuild this list as well.